Extension/src/LanguageServer/extension.ts

```diff
@@ -1379,3 +1382,88 @@ export async function getIncludes(maxDepth: number): Promise<any> {
+async function onCopilotHover(): Promise<void> {
+    if (!vscode.window.activeTextEditor) { return; }
+    // Check if the user has access to vscode language model.
+    const vscodelm = (vscode as any).lm;
+    if (!vscodelm) { return; }
+
+    // Prep hover with wait message and get the hover position location.
+    const copilotHoverResult = await clients.ActiveClient.showCopilotHover('$(loading~spin)');
+    const hoverPosition = new vscode.Position(copilotHoverResult.hoverPos.line, copilotHoverResult.hoverPos.character);
+
+    // Make sure the editor has focus.
+    await vscode.window.showTextDocument(vscode.window.activeTextEditor.document, { preserveFocus: false, selection: new vscode.Selection(hoverPosition, hoverPosition) });
+
+    // Workaround to force the editor to update it's content, needs to be called from another location first.
+    await vscode.commands.executeCommand('cursorMove', { to: 'right' });
+    await vscode.commands.executeCommand('editor.action.showHover', { focus: 'noAutoFocus' });
+
+    // Move back and show the correct hover.
+    await clients.ActiveClient.showCopilotHover('$(loading~spin)');
+    await vscode.commands.executeCommand('cursorMove', { to: 'left' });
+    await vscode.commands.executeCommand('editor.action.showHover', { focus: 'noAutoFocus'});
+
+    // Gather the content for the query from the client.
+    const response = await clients.ActiveClient.getCopilotHoverInfo();
+
+    // Ensure the content is valid before proceeding.
+    const request = response.content;
+
+    if (request.length === 0) {
+        return;
+    }
+
+    const locale = getLocaleId();
+
+    const messages = [
+        vscode.LanguageModelChatMessage
+            .User(request + locale)];
+
+    const [model] = await vscodelm.selectChatModels(modelSelector);
+
+    let chatResponse: vscode.LanguageModelChatResponse | undefined;
+    try {
+        chatResponse = await model.sendRequest(
+            messages,
+            {},
+            new vscode.CancellationTokenSource().token
+        );
+    } catch (err) {
+        if (err instanceof vscode.LanguageModelError) {
+            console.log(err.message, err.code, err.cause);
+        } else {
+            throw err;
+        }
+        return;
+    }
+
+    // Ensure we have a valid response from Copilot.
+    if (!chatResponse) { return; }
+
+    let content: string = '';
+
+    try {
+        for await (const fragment of chatResponse.text) {
+            content += fragment;
+        }
+    } catch (err) {
+        return;
+    }
+
+    if (!vscode.window.activeTextEditor) { return; }
+    await vscode.window.showTextDocument(vscode.window.activeTextEditor.document, { preserveFocus: false, selection: new vscode.Selection(hoverPosition, hoverPosition) });
+
+    // Same workaround as above to force the editor to update it's content.
+    await clients.ActiveClient.showCopilotHover('$(loading~spin)');
```

Comment:
It would be nice to have a way to do this other than wiggling the mouse.

VS Code generally merges results from multiple providers.  I suspect if there are multiple HoverProviders and only one returns results (yet) it will show only one result.  If that result had the link in it, and that link 'unblocked' a secondary HoverProvider from returning an AI result, I suspect VS Code would then populate a new 'section' of the tooltip with the additional provider result.  That is just a guess, I haven't tried it.  :)

Reply:
I finally got a chance to try this out, and it almost works perfectly, but unfortunately while the provider is blocking waiting for the signal, VS Code reserves hover space for it and displays "Loading..." until it's finally unblocked and the Promise resolves.

If you have any ideas on a way to not show the "Loading..." message, I think everything else should work. So far everything I've tried has hit either that same issue or the issue of the Hover content not updating after being rendered.

Reply:
Maybe we could put in a request to VS Code, to make this more perfect? :)  Since this is AI related, they might consider it a priority.  Hypothetically, if there were N hover providers, does it make sense for VS Code to show N `Loading...` messages?  Seems like just one would suffice, and additional results could appear for additional providers, once they provide results. That might be easy for them to fix. I'd suggest opening an issue against VS Code and seeing what they think.  https://github.com/microsoft/vscode/issues

Reply:
This latest iteration is still using the multiple invocations to update the content, but swapping to using the new Hover provider has made it a lot easier to track between requests. If you or the team still have concerns about this implementation approach, I'm happy to discuss further and we can make a call on next steps.